examples/atari/src/atari_wrappers.py

```diff
@@ -304,6 +305,83 @@ def __getitem__(self, i):
         return self._force()[..., i]
 
 
+class RawObservationWrapper(Wrapper):
+    def __init__(
+            self,
+            env,
+            allow_early_resets=False,
+            reset_keywords=(),
+            info_keywords=()
+    ):
+        Wrapper.__init__(self, env=env)
+        self.reset_keywords = reset_keywords
+        self.info_keywords = info_keywords
+        self.allow_early_resets = allow_early_resets
+        self.needs_reset = True
+        self.observations = []
+        self.actions = []
+        self.rewards = []
+        self.dones = []
+
+    def reset(self, **kwargs):
+        self.reset_state()
+        for k in self.reset_keywords:
+            v = kwargs.get(k)
+            if v is None:
+                raise ValueError(
+                    "Expected you to pass kwarg %s into reset" % k
+                )
+            self.current_reset_info[k] = v
+        initial_state = self.env.reset(**kwargs)
+        self.observations = [initial_state, ]
+        self.actions = []
+        self.rewards = []
+        self.dones = []
+        return self.env.reset(**kwargs)
+
+    def reset_state(self):
+        if not self.allow_early_resets and not self.needs_reset:
+            raise RuntimeError(
+                "Wrap your env with "
+                "RawObservationWrapper(env, allow_early_resets=True)"
+            )
+        self.needs_reset = False
+
+    def step(self, action):
+        if self.needs_reset:
+            raise RuntimeError("Tried to step environment that needs reset")
+        ob, rew, done, info = self.env.step(action)
+        self.update(ob, rew, done, action, info)
+        return (ob, rew, done, info)
+
+    def update(self, ob, rew, done, action, info):
+        self.observations.append(ob)
+        self.actions.append(action)
+        self.rewards.append(rew)
+        self.dones.append(done)
+        assert isinstance(info, dict)
+        if done:
+            self.needs_reset = True
+            info["raw_trajectory"] = (
```

Comment:
btw, I think, it would be easier to understand if you would return `info`.
inner update could cause a lot of misunderstanding